Solutions/GoogleCloudPlatformIAM/Analytic Rules/GCPIAMNewAuthenticationToken.yaml

```diff
@@ -0,0 +1,31 @@
+id: 80e4db30-5636-4fbd-8816-24c3ded8d243
+name: GCP IAM - New Authentication Token for Service Account
+description: |
+  'Detects when new authentication token is created for service account.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: GCPIAMDataConnector
+    dataTypes:
+      - GCP_IAM
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - LateralMovement
+relevantTechniques:
+  - T1550
+query: |
+  GCP_IAM
+  | where PayloadMethodname =~ 'google.iam.admin.v1.GenerateAccessToken'
+  | extend result = parse_json(todynamic(PayloadAuthorizationinfo))[0]['granted']
+  | where result =~ 'true'
+  | extend service_account = extract(@'serviceAccounts\/(.*?)@', 1, PayloadResponseName)
+  | project-away result
+  | extend AccountCustomEntity = service_account

```

Comment:
Please add any other entity types like  IPCustomEntity if it is surfaced in the log. Can we also have the timestamp field added.